node/src/reactor.rs

```diff
@@ -283,12 +299,24 @@ where
         let crank_span = debug_span!("crank", ev = self.event_count);
         let _inner_enter = crank_span.enter();
 
-        self.event_count += 1;
         self.metrics.events.inc();
 
         let event_queue = EventQueueHandle::new(self.scheduler);
         let effect_builder = EffectBuilder::new(event_queue);
 
+        // Update metrics like memory usage.
+        if self.event_count % self.event_metrics_threshold == 0 {
+            let now = Instant::now();
+
+            // We update metrics on the first very event as well to get a good baseline.
+            if self.event_count == 0
```

Comment:
Would be better to swap the order of these comparisons, since `self.event_count == 0` will always be false except the first time it's called.